docs/LibraryEvolution.md

```diff
@@ -0,0 +1,763 @@
+for the lifetime of that entity; for a global or static constant, this
+lasts from when the constant is first accessed (and lazily initialized)
+until the end of program execution. However, different versions of the
+same library may choose to have different values for a constant---say, a
+string describing the library's copyright information.
+
+In order to make use of a constant's value across library boundaries,
+the library owner may mark the constant as `@inlineable`. As when
+applied to functions, the attribute takes a version number specifying
+which versions of the library will behave correctly if the value is
+inlined into client code.
+
+Note that if the constant's initial value expression has any observable
+side effects, including the allocation of class instances, it must not
+be treated as inlineable. A constant must always behave as if it is
+initialized exactly once.
+
+> **TODO**
+>
+> Is this a condition we can detect at compile-time? Do we have to be
+> restricted to things that can be lowered to compile-time constants?
+
+Properties
+----------
+
+By default, a stored property in a struct or class may be replaced by a
+computed property in later versions of a library. As shown above, the
+`@fixed_layout` attribute promises that all stored properties currently
+in a type will remain stored in all future library versions, but
+sometimes that isn't a reasonable promise. In this case, a library owner
+may still want to allow clients to rely on a *specific* stored property
+remaining stored, by applying the `@fixed` attribute to the property.
+
+> **TODO**
+>
+> Is it valid for a fixed property to have observing accessors, or is it
+> more useful to promise that the setter is just a direct field access
+> too? If it were spelled `@fragile`, I would assume that accessors are
+> permitted but they become inlineable, and so not having any accessors
+> is just a degenerate case of that.
+>
+> Is this feature sufficiently useful to be proposed initially at all,
+> or is it too granular?
+
+Like all other attributes in this section, the `@fixed` attribute must
+specify in which version of the library clients may rely on the property
+being stored. The attribute may not be applied to non-final properties
+in classes.
+
+> Note: It would be possible to allow `@fixed` on non-final properties, and
+> have it only apply when the client code is definitively working with
+> an instance of the base class, not any of its subclasses. But this is
+> probably too subtle, and makes it look like the attribute is doing
+> something useful when it actually isn't.
+
+Other Promises About Types
+--------------------------
+
+Advanced users may want to promise more specific things about various
+types. These are similar to the internal `effects` attribute we have for
+functions, except that they can be enforced by the compiler.
+
+-   `trivial`: Promises that the type is trivial. Note that this is not a
```

Comment:
Sorry, I just meant the attribute name didn't need to be linked. The word "trivial" in the first sentence should still be linked to `#trivial`.